src/components/ContentHighlights/HighlightStepper/ContentHighlightStepper.jsx

```diff
@@ -95,103 +106,149 @@ const ContentHighlightStepper = ({ enterpriseId }) => {
       setIsPublishing(false);
     }
   };
+  const closeStepper = () => {
+    open();
+  };
+
+  /* Start beforeunload event */
+  // This section triggers browser response to unsaved items when the stepper modal is open/active
+  useEffect(() => {
+    const preventUnload = () => {
+      /* eslint-disable no-restricted-globals */
+      /* Mandatory requirements to trigger response by browser, event.preventDefault && event.returnValue
+      A return value is required to trigger the browser unsaved data blocking modal response */
+      event.preventDefault();
+      event.returnValue = 'Are you sure? Your data will not be saved.';
```

Comment:
Bummer that browsers don't let you customize the actual message that shows, but I suppose it makes sense for security reasons.